core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -100,6 +100,11 @@ private[spark] class TaskSetManager(
   // should not resubmit while executor lost.
   private val killedByOtherAttempt = new HashSet[Long]
 
+  // Add the tid of task into this HashSet when the task is killed by other stage retries.
+  // For example, if stage failed and retry, when the task in the origin stage finish, it will
+  // kill the new stage task running the same partition data
+  private val killedByOtherStageRetries = new HashSet[Long]
```

Comment:
Why is this hashset not looked up in case of handling `executorLost` ?

Reply:
Also the part of code that you marked as problematic in `executorLost` , could it have not been moved to `handleFailedTask`? I feel the code could have looked more clearer there and then rest of the changes might not have been needed.

Reply:
I have handled it in `executorLost `, thank you!
The rest of code is going to kill running tasks when it has succeeded in another stage to decrease execution time and resource.